src/main/java/gomedic/model/activity/UniqueActivityList.java

```diff
@@ -0,0 +1,165 @@
+    }
+
+    /**
+     * Returns true if the list contains conflicting activity.
+     */
+    public boolean containsConflicting(Activity toCheck) {
+        requireNonNull(toCheck);
+        if (contains(toCheck)) {
+            return internalList.stream().filter(toCheck::isConflicting).count() > 1;
+        } else {
+            return internalList.stream().anyMatch(toCheck::isConflicting);
+        }
+    }
+
+    /**
+     * Adds an activity to the list.
+     * The person must not already exist in the list.
+     */
+    public void add(Activity toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateActivityFoundException();
+        }
+
+        if (containsConflicting(toAdd)) {
+            throw new ConflictingActivityException();
+        }
+
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Removes the equivalent activity from the list.
+     * The activity must exist in the list.
+     */
+    public void remove(Activity toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new ActivityNotFoundException();
+        }
+    }
+
+    public void setActivities(UniqueActivityList replacement) {
+        requireNonNull(replacement);
+        internalList.setAll(replacement.internalList);
+    }
+
+    /**
+     * Replaces the contents of this list with {@code activities}.
+     * {@code activities} must not contain duplicate and conflicting activities.
+     */
+    public void setActivities(List<Activity> activities) {
+        CollectionUtil.requireAllNonNull(activities);
+        if (!checkActivitiesAreUnique(activities)) {
+            throw new DuplicateActivityFoundException();
+        }
+
+        if (!checkActivitiesAreConflicting(activities)) {
+            throw new ConflictingActivityException();
+        }
+
+        internalList.setAll(activities);
+    }
+
+    @Override
+    public Iterator<Activity> iterator() {
+        return internalList.iterator();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof UniqueActivityList // instanceof handles nulls
+                && internalList.equals(((UniqueActivityList) other).internalList));
+    }
+
+    /**
+     * Returns the backing list as an unmodifiable {@code ObservableList}.
+     */
+    public ObservableList<Activity> asUnmodifiableObservableList() {
+        return internalUnmodifiableList;
+    }
+
+    /**
+     * Returns the backing list as an unmodifiable {@code ObservableList}.
+     * Returned list is sorted by starting the start time.
+     */
+    public ObservableList<Activity> asUnmodifiableSortedList() {
+        return internalUnmodifiableList
+                .sorted((activity, otherAct) -> activity
+                        .getStartTime()
+                        .isBefore(otherAct.getStartTime()) ? 0 : 1);
```

Comment:
Should we use `-1` instead of `0` instead, since it makes more semantical sense w.r.t the expected behaviour stated in Java's `Comparator::compare` contract?
```suggestion
                        .isBefore(otherAct.getStartTime()) ? -1 : 1);
```

Reply:
Yeah sure, ill add it because apparently I need to return 0 for equality